analyses/cms-open-data-ttbar/statistical.py

```diff
@@ -0,0 +1,471 @@
+        histoname_down="4j2b_single_top_t_chan_btag_var_2_down",
+    )
+    single_top_t_chan.AddNormPlusShapeHistoSys(
+        "b_tag_NP_4",
+        histoname_up="4j2b_single_top_t_chan_btag_var_3_up",
+        histoname_down="4j2b_single_top_t_chan_btag_var_3_down",
+    )
+
+    channel_2b.AddSample(single_top_t_chan)
+
+    single_top_tW = AGC_Sample(
+        "single_top_tW", "4j2b_single_top_tW", input_file
+    )
+    single_top_tW.ActivateStatError()
+    single_top_tW.SetSystematicsInputFile(input_file)
+    single_top_tW.AddOverallSys(lumi_systematics)
+    single_top_tW.AddNormPlusShapeHistoSys(
+        "jet_energy_scale", histoname_up="4j2b_single_top_tW_pt_scale_up"
+    )
+    single_top_tW.AddNormPlusShapeHistoSys(
+        "jet_energy_res", histoname_up="4j2b_single_top_tW_pt_res_up"
+    )
+    single_top_tW.AddNormPlusShapeHistoSys(
+        "b_tag_NP_1",
+        histoname_up="4j2b_single_top_tW_btag_var_0_up",
+        histoname_down="4j2b_single_top_tW_btag_var_0_down",
+    )
+    single_top_tW.AddNormPlusShapeHistoSys(
+        "b_tag_NP_2",
+        histoname_up="4j2b_single_top_tW_btag_var_1_up",
+        histoname_down="4j2b_single_top_tW_btag_var_1_down",
+    )
+    single_top_tW.AddNormPlusShapeHistoSys(
+        "b_tag_NP_3",
+        histoname_up="4j2b_single_top_tW_btag_var_2_up",
+        histoname_down="4j2b_single_top_tW_btag_var_2_down",
+    )
+    single_top_tW.AddNormPlusShapeHistoSys(
+        "b_tag_NP_4",
+        histoname_up="4j2b_single_top_tW_btag_var_3_up",
+        histoname_down="4j2b_single_top_tW_btag_var_3_down",
+    )
+
+    channel_2b.AddSample(single_top_tW)
+    meas.AddChannel(channel_2b)
+
+    meas.SetPOI("ttbar_norm")
+    meas.CollectHistograms()
+    meas.SetExportOnly(True)
+
+    ws = ROOT.RooStats.HistFactory.MakeModelAndMeasurementFast(meas)
+
+    # Retrieve the ModelConfig
+    modelConfig = ws.obj("ModelConfig")
+
+    # Extract the PDF and global observables
+    pdf = modelConfig.GetPdf()
+    globalObservables = ROOT.RooArgSet(modelConfig.GetGlobalObservables())
+
+    # Perform the fit
+    result = pdf.fitTo(
+        ws.data("obsData"),
+        ROOT.RooFit.Save(),
+        ROOT.RooFit.PrintLevel(-1),
+        ROOT.RooFit.GlobalObservables(globalObservables),
+    )
+
+    vis = Visualization()
+    vis.CreateAndSaveResultsPicture(
+        "fitted_parameters.png", ws
+    )  # Parameters fit result
+
+    vis.DrawCorrelationMatrix(
+        "correlation_matrix.png", result
+    )  # Correlation matrix
+
+    md = DrawModel(meas, ws)
+    md.Draw(result)
+
+    md.GetChannelPrefitGraph(0).SetTitle(
+        "4j1b_CR PREFIT GRAPH"
+    )  # each graph can be accessed from outside and modified
+    md.GetChannelPrefitGraph(0).GetYaxis().SetTitle("number of entries")
+
+    # save fit results to ROOT file
+
+    file = ROOT.TFile("fitResults.root", "RECREATE")
+
+    result.Write("fitResult")
+
+    file.Close()
```

Comment:
```suggestion
    with ROOT.TFile.Open("fitResults.root", "RECREATE") as file:
        result.Write("fitResult")
```
We're promoting the use of contexts for file handling, so that the user can't accidentally forget to close the file. I would also do this here.